Model/Api/DiscountCodeValidation.php

```diff
@@ -272,7 +272,10 @@ private function applyingGiftCardCode($code, $giftCard, $immutableQuote, $parent
     private function getCartTotals($quote)
     {
         $is_has_shipment = !empty($this->requestArray['cart']['shipments'][0]['reference']);
-        $cart = $this->cartHelper->getCartData($is_has_shipment, null, $quote);
+        $payload = isset($this->requestArray['cart']['billing_address'])
```

Comment:
Can we add this code under the feature switch so we can simply revert it once we delete the feature?
Otherwise, it just adds more complexity to the plugin.

Reply:
Please check again